### ISSUE TITLE
Feature Request : EditBox

### ISSUE THREAD

**Issue by yesnault** [gizak/termui#34]
Hi,

Is there an easy way to make an edit box with termui ?

Tks,

**Comment by gizak** [comment]
I think for now you have to do:

1. listen keyboard events
2. get every pressed key stored
3. put the captured char into the dataset of a component
4. refresh 

`Input` widget is planed but we are not getting there yet, since there is an undergoing refactoring and only after this I will switch to the event system and new widgets :)

**Comment by weberc2** [comment]
+1

**Comment by mikepea** [comment]
I've got a basic implementation of a termui-based editbox working in https://github.com/mikepea/go-jira-ui

It's basically a port of the EditBox example code from termbox-go.

It's not in a state where it'd be contributable upstream (yet), but it'd be a reasonable base to use in your own projects. See ui_controls.go for how it's connected to the keyboard event handler.

**Comment by liushuchun** [comment]
yeah.I need this feature too.it's not easy to get the input text string in termui.

**Comment by brianbianco** [comment]
Any info of if this is still in the roadmap or not? I think this is a pretty essential feature.

**Comment by liushuchun** [comment]
I am also waiting for this basic feature.

**Comment by liushuchun** [comment]
hi all,I have solved this problem.
https://github.com/liushuchun/wechatcmd

**Comment by SAPessi** [comment]
Hi all,

I have created a basic input component for termui, you can see it in my fork here: https://github.com/SAPessi/termui/commit/71d140f624b87461a4ed362e55fc2d1c13c0c7a4

You can use it like this:
```go
// first param is the string content, the second whether it should be multi-line or not
input := termui.NewInput("", true)
input.Height = termui.TermHeight()
input.BorderFg = termui.ColorYellow
// same properties as other Block
termui.Render(input)

// the StartCapture method tells the editor to begin listening for /sys/kbd events
input.StartCapture()
```

@gizak interested in a pull request?

**Comment by gizak** [comment]
@SAPessi Hi pull requests are very welcome!

**Comment by SAPessi** [comment]
Sent as pull requests #129

**Comment by max123926** [comment]
> Hi all,
> 
> I have created a basic input component for termui, you can see it in my fork here: [sapessi@71d140f](https://github.com/sapessi/termui/commit/71d140f624b87461a4ed362e55fc2d1c13c0c7a4)
> 
> You can use it like this:
> 
> ```go
> // first param is the string content, the second whether it should be multi-line or not
> input := termui.NewInput("", true)
> input.Height = termui.TermHeight()
> input.BorderFg = termui.ColorYellow
> // same properties as other Block
> termui.Render(input)
> 
> // the StartCapture method tells the editor to begin listening for /sys/kbd events
> input.StartCapture()
> ```
> 
> @gizak interested in a pull request?
> 
> **Update**
> Made a few changes since the original commit, take a look at the file here: https://github.com/SAPessi/termui/blob/master/input.go
> 
> Also working on a dropdown component

Excuse me, I put "input.go" in "$GOPATH\src\github.com\gizak\termui\v3".
However, there show some errors:
undefined: Attribute go [53,15]
undefined: Attribute go [54,15]
undefined: TextBuilder go [57,15]

How could I edit the errors?
Thank you very much!
Max

**Comment by max123926** [comment]
> Hi all,
> 
> I have created a basic input component for termui, you can see it in my fork here: [sapessi@71d140f](https://github.com/sapessi/termui/commit/71d140f624b87461a4ed362e55fc2d1c13c0c7a4)
> 
> You can use it like this:
> 
> ```go
> // first param is the string content, the second whether it should be multi-line or not
> input := termui.NewInput("", true)
> input.Height = termui.TermHeight()
> input.BorderFg = termui.ColorYellow
> // same properties as other Block
> termui.Render(input)
> 
> // the StartCapture method tells the editor to begin listening for /sys/kbd events
> input.StartCapture()
> ```
> 
> @gizak interested in a pull request?
> 
> **Update**
> Made a few changes since the original commit, take a look at the file here: https://github.com/SAPessi/termui/blob/master/input.go
> 
> Also working on a dropdown component

Excuse me, I use go version 12.9 on Windows 10.
I put "input.go" in "$GOPATH\src\gopkg.in\gizak\termui.v2" and ran the example you offered.
However, it occurred fatal error.
I've posted the problem [here](https://groups.google.com/forum/?hl=zh-TW#!topic/golang-nuts/dcpEIVTWZKw).

Thank you very much!
Max
